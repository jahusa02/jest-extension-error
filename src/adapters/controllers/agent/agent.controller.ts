import { CommonRoutesConfig } from '../base/common/common-routes-config';
import { NextFunction, Request, Response, Router } from 'express';
import { GetAgentUseCase } from '../../../app/modules/agent/use-cases/get-agent/get-agent';
import * as getAgentModels from '../../../app/modules/agent/use-cases/get-agent/get-agent.models';
import RouteNotImplementedError from '../../../app/shared/errors/route-not-implemented-error';

export class AgentController extends CommonRoutesConfig {
  protected readonly name = 'AgentController';

  constructor(private getAgentUseCase: GetAgentUseCase) {
    super();
  }

  configureRoutes(router: Router): Router {
    router.get('/agent', this.getAgentHandler.bind(this));
    router.get('/agent*', (_request: Request, _response: Response, next: NextFunction) => {
      next(new RouteNotImplementedError());
    });

    return router;
  }

  private getAgentHandler(_request: Request, response: Response, next: NextFunction): void {
    const requestModel: getAgentModels.RequestModel = {
      user: {
        id: '33399',
      },
    };

    void this.getAgentUseCase
      .execute(requestModel)
      .then((result: getAgentModels.ResponseModel) => {
        response.status(200).json(result);
      })
      .catch(next);
  }
}
