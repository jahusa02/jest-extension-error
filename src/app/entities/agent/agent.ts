export interface AgentEntityDto {
  id: string;
  partnerNumber: string;
  agentNumber: string;
  firstName: string;
  lastName: string;
  locale: string;
}

export class AgentEntity {
  constructor(private data: AgentEntityDto) {}

  getId(): string {
    return this.data.partnerNumber;
  }

  getPartnerNumber(): string {
    return this.data.partnerNumber;
  }

  getAgentNumber(): string {
    return this.data.agentNumber;
  }

  getFirstName(): string {
    return this.data.firstName;
  }

  getLastName(): string {
    return this.data.lastName;
  }

  getLocale(): string {
    return this.data.locale.toLowerCase() + '-de';
  }
}
