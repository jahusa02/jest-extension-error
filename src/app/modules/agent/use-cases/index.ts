import { GetAgentUseCase } from './get-agent/get-agent';

const makeGetAgentUseCase = (): GetAgentUseCase => {
  return new GetAgentUseCase();
};

export const agentUseCases = {
  getAgent: makeGetAgentUseCase(),
};
