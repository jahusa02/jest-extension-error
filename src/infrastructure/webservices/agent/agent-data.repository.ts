import { WgWebServiceAdapter } from '../../utils/wg-web-service-adapter';

export interface AgentData {
  PARTNER_NUMBER: string;
  AGENT_NUMBER: string;
  NAME: string;
  NAME2: string;
}

interface WsResultAgentData {
  AGENTS: {
    AGENT: AgentData;
  };
}

export class AgentDataRepository {
  private readonly getAgentDefaultParameter = {
    FUNCTION: 'GET_AGENT',
    SHAPING: 'MULTI_WITH_EXCLUSION',
    AGENT_NUMBER: '',
    AMOUNT: 1,
    API: 'JSON',
  };

  constructor(private webServiceAdapter: WgWebServiceAdapter) {}

  getFor(userId: string): Promise<AgentData | Error> {
    const parameter = {
      ...this.getAgentDefaultParameter,
      AGENT_NUMBER: userId,
    };

    return this.webServiceAdapter
      .call<WsResultAgentData>(parameter)
      .then((responseBody: WsResultAgentData) => {
        return responseBody.AGENTS.AGENT;
      })
      .catch((error: Error) => {
        return error;
      });
  }
}
