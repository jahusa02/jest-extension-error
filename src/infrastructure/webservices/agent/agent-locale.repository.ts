import { WgWebServiceAdapter } from '../../utils/wg-web-service-adapter';

export interface AgentLocale {
  COUNTRY: string;
}

interface WsResultAgentLocale {
  ADDRESS: AgentLocale;
}

export class AgentLocaleRepository {
  private readonly getAddressDefaultParameter = {
    FUNCTION: 'GET_ADDRESS',
    SHAPING: 'BY_PARTNER',
    PARTNER_NR: '',
    AMOUNT: 1,
    PAGE: 1,
    API: 'JSON',
  };

  constructor(private webServiceAdapter: WgWebServiceAdapter) {}

  getFor(partnerNumber: string): Promise<AgentLocale | Error> {
    const parameter = {
      ...this.getAddressDefaultParameter,
      PARTNER_NR: partnerNumber,
    };

    return this.webServiceAdapter
      .call<WsResultAgentLocale>(parameter)
      .then((responseBody: WsResultAgentLocale) => responseBody.ADDRESS)
      .catch((error: Error) => {
        return error;
      });
  }
}
