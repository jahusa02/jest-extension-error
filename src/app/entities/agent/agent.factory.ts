import { AgentEntity, AgentEntityDto } from './agent';

export class AgentEntityFactory {
  create(options: AgentEntityDto): AgentEntity {
    return new AgentEntity(options);
  }
}
