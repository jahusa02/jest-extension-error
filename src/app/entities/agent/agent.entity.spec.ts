import { AgentEntityFactory } from './agent.factory';
import { AgentEntity, AgentEntityDto } from './agent';

describe('agent entity', () => {
  let factory: AgentEntityFactory;

  const options: AgentEntityDto = {
    id: 'myId',
    agentNumber: '0123',
    firstName: 'Foo',
    lastName: 'Bar',
    locale: 'de',
    partnerNumber: '3456',
  };

  beforeEach(() => {
    factory = new AgentEntityFactory();
  });

  describe('AgentEntityFactory', () => {
    it('should be created', () => {
      expect(factory).toBeDefined();
    });

    it('should return an instance of AgentEntity', () => {
      expect(factory.create(options)).toBeInstanceOf(AgentEntity);
    });
  });

  describe('AgentEntity', () => {
    it('should contain given options', () => {
      // Arrange
      const entity: AgentEntity = factory.create(options);
      // Act
      const data = {
        id: entity.getId(),
        agentNumber: entity.getAgentNumber(),
        firstName: entity.getFirstName(),
        lastName: entity.getLastName(),
        locale: entity.getLocale(),
        partnerNumber: entity.getPartnerNumber(),
      };
      // Assert
      expect(data).toMatchSnapshot();
    });
  });
});
