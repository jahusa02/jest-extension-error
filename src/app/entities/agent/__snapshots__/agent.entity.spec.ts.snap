// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`agent entity AgentEntity should contain given options 1`] = `
Object {
  "agentNumber": "0123",
  "firstName": "Foo",
  "id": "3456",
  "lastName": "Bar",
  "locale": "de-de",
  "partnerNumber": "3456",
}
`;
